cmd/state-offline-installer/install.go

```diff
@@ -0,0 +1,291 @@
+		return errs.Wrap(err, "Unable to create artifactsPath directory")
+	}
+
+	r.out.Print(fmt.Sprintf("Stage 2 of 3 Start: Decompressing artifacts into: %s", artifactsPath))
+	archivePath := filepath.Join(assetsPath, artifactsTarGZName)
+	ua := unarchiver.NewTarGz()
+	f, siz, err := ua.PrepareUnpacking(archivePath, artifactsPath)
+	if err != nil {
+		return errs.Wrap(err, "Unable to prepare unpacking of artifact tarball")
+	}
+
+	ua.SetNotifier(func(filename string, _ int64, isDir bool) {
+		if !isDir {
+			r.out.Print(fmt.Sprintf("Unpacking artifact %s", filename))
+		}
+	})
+
+	err = ua.Unarchive(f, siz, artifactsPath)
+	if err != nil {
+		return errs.Wrap(err, "Unable to unarchive artifacts to artifactsPath")
+	}
+
+	r.out.Print(fmt.Sprintf("Stage 2 of 3 Finished: Decompressing artifacts into: %s", artifactsPath))
+
+	return nil
+}
+
+func (r *runner) extractAssets(assetsPath string, backpackZipFile string) error {
+	if err := os.Mkdir(assetsPath, os.ModePerm); err != nil {
+		return errs.Wrap(err, "Unable to create assetsPath")
+	}
+
+	ua := unarchiver.NewZip()
+	r.out.Print(fmt.Sprintf("Stage 1 of 3 Start: Decompressing assets into: %s", assetsPath))
+	f, siz, err := ua.PrepareUnpacking(backpackZipFile, assetsPath)
+	if err != nil {
+		return errs.Wrap(err, "Unable to prepare unpacking of backpack")
+	}
+
+	err = ua.Unarchive(f, siz, assetsPath)
+	if err != nil {
+		return errs.Wrap(err, "Unable to unarchive Assets to assetsPath")
+	}
+
+	r.out.Print(fmt.Sprintf("Stage 1 of 3 Finished: Decompressing assets into: %s", assetsPath))
+	return nil
+}
+
+func (r *runner) configureEnvironment(path string, asrt *runtime.Runtime) error {
+	configureEnvironmentAccepted, err := r.prompt.Confirm("Setup", "Setup environment for installed project?", p.BoolP(true))
+	if err != nil {
+		return errs.Wrap(err, "Error getting confirmation")
+	}
+
+	if !configureEnvironmentAccepted {
+		return nil
+	}
+	env, err := asrt.Env(false, false)
+	if err != nil {
+		return errs.Wrap(err, "Error setting environment")
+	}
+
+	if rt.GOOS == "windows" {
+		contents, err := assets.ReadFileBytes("scripts/setenv.bat")
+		if err != nil {
+			return errs.Wrap(err, "Error reading file bytes")
+		}
+		err = fileutils.WriteFile(filepath.Join(path, "setenv.bat"), contents)
+		if err != nil {
+			return locale.WrapError(err,
+				"err_deploy_write_setenv",
+				"Could not create setenv batch scriptfile at path: {{.V0}}",
+				path)
+		}
+	}
+
+	err = r.shell.WriteUserEnv(r.cfg, env, sscommon.OfflineInstallID, true)
+	if err != nil {
+		return locale.WrapError(err,
+			"err_deploy_subshell_write",
+			"Could not write environment information to your shell configuration.")
+	}
+
+	binPath := filepath.Join(path, "bin")
+	if err := fileutils.MkdirUnlessExists(binPath); err != nil {
+		return locale.WrapError(err, "err_deploy_binpath", "Could not create bin directory.")
+	}
+
+	// Write global env file
+	err = r.shell.SetupShellRcFile(binPath, env, nil)
```

Comment:
Note I dropped the namespace param here and refactored `SetupShellRCFile` to not alter the PS1 if no namespace was provided.